helm/templates/solr.yml

```diff
@@ -0,0 +1,59 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: sdap-solr
+spec:
+  type: ClusterIP
+  ports:
+  - port: 8983
+    protocol: TCP
+    targetPort: http
+  selector:
+    app: sdap-solr
+
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: solr-set
+spec:
+  selector:
+    matchLabels:
+      app: sdap-solr # has to match .spec.template.metadata.labels
+  serviceName: "sdap-solr"
+  replicas: {{ .Values.solr.replicas }}
```

Comment:
I don't think there is a valid use case where you would deploy more than 1 replica of the `solr-singlenode` image. This would simply result in multiple segregated solr nodes that know nothing of each other but would be attached behind the solr service proxy meaning any service could respond to any request. So a document could end up being indexed into replica-1 but requested from replica-2, etc...

You'd have to use the https://github.com/apache/incubator-sdap-nexus/tree/master/docker/solr/cloud image along with the https://github.com/apache/incubator-sdap-nexus/tree/master/docker/solr/cloud-init to properly setup a solr cloud cluster.